src/main.tsx

```diff
@@ -1,10 +1,61 @@
-import React from 'react'
-import ReactDOM from 'react-dom/client'
-import App from './App.tsx'
-import './style/style.scss'
+import React from "react";
+import ReactDOM from "react-dom/client";
+import "./style/style.scss";
 
-ReactDOM.createRoot(document.getElementById('root')!).render(
+import { createBrowserRouter, RouterProvider } from "react-router-dom";
+import { MainPage } from "./component/pages/main-page/main-page.tsx";
+import ErrorPage from "./pages/error-page/error-page.tsx";
+import CategoryShop from "./pages/category-shop/category-shop.tsx";
+import CartShop from "./pages/cart-shop/cart-shop.tsx";
+import CheckoutShop from "./pages/checkout-shop/checkout-shop.tsx";
+import OrderConfirmationShop from "./pages/order-confirmation-shop/order-confirmation-shop.tsx";
+import ProductShop from "./pages/product-shop/product-shop.tsx";
+import { apiGetCategories, CategoriesModel } from "./api/api-get-categories.ts";
+
+export async function loader() {
+  const category : CategoriesModel[] | undefined = await apiGetCategories("https://lopi2-backend-5517f8f04d28.herokuapp.com/api/");
+  return category ;
+} 
+
+const router = createBrowserRouter([
```

Comment:
It is going to expand very soon. Let's move it to a separate file. You could create another directory like `routes`

Reply:
ok